activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb

```diff
@@ -1018,11 +1018,16 @@ def prevent_writes=(prevent_writes) # :nodoc:
         Thread.current[:prevent_writes] = prevent_writes
       end
 
-      # Prevent writing to the database regardless of role.
+      # Prevent regular writing to the database regardless of role.
       #
       # In some cases you may want to prevent writes to the database
       # even if you are on a database that can write. `while_preventing_writes`
-      # will prevent writes to the database for the duration of the block.
+      # will prevent most writes to the database for the duration of the block.
+      #
+      # Note: `while_preventing_writes` does not necessarily catch all writes,
+      # it's meant as a safeguard against accidental writes. Writes as created
+      # by ActiveRecord will most likely be caught. Handcrafted write queries
+      # may still be possible though.
```

Comment:
This statement makes it sound like there isn't an explicit list of the kinds of queries that writes are prevented for. Can you write this more clearly and reference the list of disallowed queries?

Reply:
Well there really isn't any write query you can't do in Postgresql, you can make them all run as I pointed out in the issue #39133. Or to say it differently, there aren't any disallowed queries when you prepend your query with a `/*/**/SELECT*/` comment.

Reply:
If you want true replicas you should use a readonly user on your replica. Rails can only do so much guess work for you.

The new documentation is just not clear enough to be something we can merge. All the dbs have an explicit list, even if there are ways around that. The new docs make it sound like Rails just blocks writes when it feels like it, but it's really more about how you craft your queries.

Reply:
OK, maybe we can add something along the lines of: 

> Note: `while_preventing_writes` is not as safe as a separate readonly database user and it does not make any guarantees against SQL injections, it is meant to provide a safeguard against accidental writes by the Rails application itself.